This patch fixes a mistake in the buffer_time parameter of alsa09 driver.
The driver used to consider it in microseconds, but the rest of libao
and the documentation declare it in milliseconds.

Patch by Heikki Orsila.

Gentoo Bug #119967
Upstream ticket http://trac.xiph.org/ticket/814
Index: libao-0.8.6/src/plugins/alsa09/ao_alsa09.c
===================================================================
--- libao-0.8.6.orig/src/plugins/alsa09/ao_alsa09.c
+++ libao-0.8.6/src/plugins/alsa09/ao_alsa09.c
@@ -166,7 +166,7 @@ int ao_plugin_set_option(ao_device *devi
 			return 0;
 	}
 	else if (!strcmp(key, "buffer_time"))
-		internal->buffer_time = atoi(value);
+		internal->buffer_time = atoi(value) * 1000;
 	else if (!strcmp(key, "period_time"))
 		internal->period_time = atoi(value);
 	else if (!strcmp(key,"use_mmap")) {
@@ -264,7 +264,7 @@ static inline int alsa_set_hwparams(ao_a
 	if (err < 0)
 		return err;
 
-	/* set the length of the hardware sample buffer in milliseconds */
+	/* set the length of the hardware sample buffer in microseconds */
 	internal->cmd = "snd_pcm_hw_params_set_buffer_time_near";
 	err = snd_pcm_hw_params_set_buffer_time_near(internal->pcm_handle,
 			params, &(internal->buffer_time), 0);
